jOOL/src/main/java/org/jooq/lambda/Agg.java

```diff
@@ -886,6 +970,126 @@ else if (percentile == 1.0)
+     * <p>This method is different with <code>percentileAllBy</code> since this method will return the result
+     * after applying the <code>function</code>.
+     * @param <T>           The type of element
+     * @param <U>           The type of result of <code>function</code>
+     * @param percentile    The percentile number
+     * @param function      The given function
+     * @param comparator    The given comparator
+     * @return              A {@link Seq} of type <code>T</code>
+     */
+    public static <T, U> Collector<T, ?, Seq<U>> percentileAll(final double percentile, final Function<? super T, ? extends U> function, final Comparator<? super U> comparator) {
+        return collectingAndThen(percentileAllBy(percentile, function, comparator), t -> t.map(function));
+    }
+
+    /**
+     * Get a {@link Collector} that calculates the derived <code>PERCENTILE_DISC(percentile)</code> function
+     * given a natural ordering.
+     * <p>This method will sort the elements after applying the <code>function</code> based on
+     * <code>comparator</code> at first, then it will find the all eligible results.
+     * @param <T>           The type of element
+     * @param <U>           The type of result of <code>function</code>
+     * @param percentile    The percentile number
+     * @param function      The given function
+     * @return              A {@link Seq} of type <code>T</code>
+     */
+    public static <T, U extends Comparable<? super U>> Collector<T, ?, Seq<T>> percentileAllBy(final double percentile, final Function<? super T, ? extends U> function) {
+        return percentileAllBy(percentile, function, naturalOrder());
+    }
+
+    /**
+     * Get a {@link Collector} that calculates the derived <code>PERCENTILE_DISC(percentile)</code> function
+     * given a specific ordering.
+     * <p>This method will sort the elements after applying the <code>function</code> based on
+     * <code>comparator</code> at first, then it will find the all eligible results.
+     * @param <T>           The type of element
+     * @param <U>           The type of result of <code>function</code>
+     * @param percentile    The percentile number
+     * @param function      The given function
+     * @param comparator    The given comparator
+     * @return              A {@link Seq} of type <code>T</code>
+     * @throws IllegalArgumentException when the percentile is less than 0 or more than 1
+     */
+    public static <T, U> Collector<T, ?, Seq<T>> percentileAllBy(final double percentile, final Function<? super T, ? extends U> function, final Comparator<? super U> comparator) {
+        if (percentile < 0.0 || percentile > 1.0) {
+            throw new IllegalArgumentException("Percentile must be between 0.0 and 1.0");
+        }
+
+        if (percentile == zero) {
+            return minAllBy(function, comparator);
+        } else if (percentile == one) {
+            return maxAllBy(function, comparator);
+        }
```

Comment:
I like this optimisation. We should implement it also for the ordinary `percentile()` calculations: https://github.com/jOOQ/jOOL/issues/376